ivy/functional/ivy/meta.py

```diff
@@ -533,58 +533,95 @@ def fomaml_step(
-        inner loop will also be returned. variables for all tasks will be returned with
-        'all'. Default is ``False``.
-    num_tasks
+    inner_optimization_step : Callable[
+            [Callable[[ivy.Array, ivy.Container], ivy.Array], ivy.Container, float],
+            ivy.Container,
+        ], optional
+        The function used for the inner loop optimization. It takes the cost function,
+        variables, and learning rate as arguments, and returns the updated variables.
+        Default is `gradient_descent_update`.
+    batched : bool, optional
+        Whether to batch along the time dimension and run the meta steps in batch.
+        Default is `True`.
+    return_inner_v : Union[str, bool], optional
+        Either `'first'`, `'all'`, or `False`. If `'first'`, the variables for the first
+        task inner loop will also be returned. If `'all'`, variables for all tasks will
+        be returned. Default is `False`.
+    num_tasks : Optional[int], optional
         Number of unique tasks to inner-loop optimize for the meta step. Determined from
-        batch by default.
-    stop_gradients
-        Whether to stop the gradients of the cost. Default is ``True``.
+        the batch by default.
+    stop_gradients : bool, optional
+        Whether to stop the gradients of the cost. Default is `True`.
 
     Returns
     -------
-    ret
-        The cost and the gradients with respect to the outer loop variables.
+    Tuple[ivy.Array, ivy.Container, Any]
+        The cost, the gradients with respect to the outer loop variables, and additional
+        information from the inner loop optimization.
+
+    Notes
+    -----
+    The Reptile algorithm aims to approximate the optimal initialization of the
+    variables for a set of tasks. It achieves this by iteratively performing the
+    following steps:
+    1. For each task in the batch, perform a few gradient steps using the inner loop
+       optimization function.
+    2. Compute the average gradient across tasks.
+    3. Update the outer loop variables using the average gradient.
+
+    The `inner_grad_steps` and `inner_learning_rate` parameters control the number of
+    gradient steps performed and the learning rate used during the inner loop
+    optimization.
+
+    The `inner_optimization_step` parameter specifies the function used for the inner
+    loop optimization. It takes the cost function, variables, and learning rate as
+    arguments and returns the updated variables. The default function
+    `gradient_descent_update` performs gradient descent using the provided learning
+    rate.
+
+    The `return_inner_v` parameter controls which variables are returned from the inner
+    loop optimization. If set to `'first'`, the variables for the first task inner loop
+    will also be returned. If set to `'all'`, variables for all tasks will be returned.
+    By default, only the cost and gradients are returned.
```

Comment:
Please can you add some docstring examples here? You can find more info on them here: https://unify.ai/docs/ivy/overview/deep_dive/docstring_examples.html. The docstring examples from this stale reptile_step PR would be a great start point https://github.com/unifyai/ivy/pull/15120